core/server/helpers/img_url.js

```diff
@@ -17,19 +19,57 @@ module.exports = function imgUrl(attr, options) {
         return;
     }
 
-    var absolute = options && options.hash && options.hash.absolute;
+    const absolute = options && options.hash && options.hash.absolute;
+
+    const size = options && options.hash && options.hash.size;
+    const imageSizes = options && options.data && options.data.config && options.data.config.image_sizes;
+
+    const image = getImageWithSize(attr, size, imageSizes);
 
     // CASE: if attribute is passed, but it is undefined, then the attribute was
     // an unknown value, e.g. {{img_url feature_img}} and we also show a warning
-    if (attr === undefined) {
+    if (image === undefined) {
         proxy.logging.warn(proxy.i18n.t('warnings.helpers.img_url.attrIsRequired'));
         return;
     }
 
-    if (attr) {
-        return urlService.utils.urlFor('image', {image: attr}, absolute);
+    if (image) {
+        return urlService.utils.urlFor('image', {image}, absolute);
     }
 
     // CASE: if you pass e.g. cover_image, but it is not set, then attr is null!
     // in this case we don't show a warning
 };
+
+function getImageWithSize(imagePath, requestedSize, imageSizes) {
+    if (!imagePath) {
+        return imagePath;
+    }
+    if (!requestedSize) {
+        return imagePath;
+    }
+
+    if (/https?:\/\//.test(imagePath)) {
```

Comment:
@ErisDS This check here is to stop us from trying to change images when they're external e.g. unsplash, or from the fixtures. However, when we eventually default to v2 Content API being _the_ api behind rendering themes, this will break - because the Content API is going to serve absolute URL's all the time (I'm pretty sure that's the case?).

Reply:
Correct, we need to start thinking about images as being a read-only assets API that lives at `[config.url]/content/images`.

At some point, we'll need to take more steps towards implementing it that way, pulling some of the logic out of the current setup.

For now, having utilities that understand this, and can tell us whether an image is coming from the "image API" or something else, should be enough?

Reply:
> At some point, we'll need to take more steps towards implementing it that way, pulling some of the logic out of the current setup.

Agreed :)

I'm not 100% sure about the resolution for this however - do we merge it as is? Or wait until it supports the v2 content API?

Reply:
Unless I'm missing something, this check was added in this PR, IMO the code added here should be compatible with both APIs.

Reply:
Ah, right - that's not the case. I was trying to explain that this check _won't_ work with the v2 Content API - because this check essentially says, if the image url is absolute - skip over and don't do anything.

Reply:
I'm understand that it isn't compatible with the v2 API, but I don't understand this: 

> I'm not 100% sure about the resolution for this however - do we merge it as is? Or wait until it supports the v2 content API?

What is "it" in this suggestion? I assumed the helper, but don't understand who or what we are waiting for - you wrote the code 😬

Reply:
:smile: We're waiting for me! Either we wait until next monday and i add something to support Content API v2, or we merge, and raise an issue, to be done either next monday - or some point in the future.

Reply:
> IMO the code added here should be compatible with both APIs.

I read this as "currently it should work" rather than "it should be updated so it works" - my :b:

Reply:
> Either we wait until next monday and i add something to support Content API v2, or we merge, and raise an issue, to be done either next monday 

Either one of those is fine, as long as it is next Monday - given that the v2 API is a high priority item we want to avoid putting extra TODOs in the way of shipping.